langlib/lang.table/src/main/ballerina/table.bal

```diff
@@ -134,6 +249,30 @@ public isolated function forEach(table<MapType> t, @isolatedParam function(MapTy
 #
 # The resulting table will have the same keys as the argument table.
 #
+# ```ballerina
+# table<Employee> key(id) employees = table [
+#     {name: "Jo", salary: 1200, id: 2121},
+#     {name: "Emma", salary: 800, id: 2122},
+#     {name: "John", salary: 1500, id: 2123}
+# ];
+#
+# table<Employee> key<int> lowSalaryEmployees = employees.filter((employee) => employee.salary < 1000);
+#
+# table<Employee> key<int> highSalaryEmployees = 
+#   employees.filter(isolated function (Employee employee) returns boolean {
+#       if employee.salary > 1000 {
+#           return true;
+#       }
+#       return false;
+#   });
```

Comment:
Can also be just

```ballerina
table<Employee> key<int> highSalaryEmployees = employees.filter(employee => employee.salary > 1000);
```

Shall we think of a different example? We can wait until the array discussion (whether we need multiple examples like this) is finalized.